ale_linters/c/checkpatch.vim

```diff
@@ -0,0 +1,59 @@
+" Author: Matt Ihlenfield, mtihlenfield@protonmail.com
+" Description: checkpatch linter for c files
+
+let s:default_options = '--strict --no-tree'
+
+" Set the  location of the checkpatch executable. By default it assumes it
+" is in $(pwd)/scripts/checkpatch.pl
+call ale#Set('c_checkpatch_executable', '')
+
+call ale#Set('c_checkpatch_options', s:default_options)
+
+function! ale_linters#c#checkpatch#GetExecutable(buffer) abort
+    let l:checkpatch_executable = ale#Var(a:buffer, 'c_checkpatch_executable')
+
+    if l:checkpatch_executable  is# ''
+        let l:checkpatch_executable = fnamemodify('./scripts/checkpatch.pl', ':p')
+    endif
+
+    return l:checkpatch_executable
+endfunction
+
+function! ale_linters#c#checkpatch#GetCommand(buffer) abort
+    let l:executable = ale_linters#c#checkpatch#GetExecutable(a:buffer)
+    let l:required_options = ' --terse --no-summary'
+
+    return ale#Escape(l:executable)
+                \ . l:required_options . ' '
+                \ . ale#Var(a:buffer, 'c_checkpatch_options')
+                \ . ' -f %s'
```

Comment:
Use 4 space indentation for multi-line expressions and leave `\` at the same level of indentation, like in other files.